external/src/main/java/edu/illinois/cs/cogcomp/pipeline/handlers/StanfordRelationsHandler.java

```diff
@@ -0,0 +1,100 @@
+import edu.stanford.nlp.pipeline.StanfordCoreNLP;
+import edu.stanford.nlp.util.CoreMap;
+
+import java.util.*;
+
+/**
+ * Relation-mention detection with Stanford CoreNLP
+ */
+public class StanfordRelationsHandler extends Annotator {
+
+    public static final String viewName = "STANFORD_RELATIONS";
+
+    private StanfordCoreNLP pipeline = null;
+
+    public StanfordRelationsHandler() {
+        super(viewName, new String[]{}, true);
+    }
+
+    @Override
+    public void initialize(ResourceManager rm) {
+        Properties props = new Properties();
+        props.setProperty("annotators", "tokenize,ssplit,pos,lemma,ner,parse,depparse,relation");
+        this.pipeline = new StanfordCoreNLP(props);
+    }
+
+    @Override
+    protected void addView(TextAnnotation ta) throws AnnotatorException {
+        Annotation document = new Annotation(ta.text);
+        pipeline.annotate(document);
+        SpanLabelView vu = new SpanLabelView(viewName, ta);
+
+        for (CoreMap sentence : document.get(CoreAnnotations.SentencesAnnotation.class)) {
+            for(RelationMention rm : sentence.get(MachineReadingAnnotations.RelationMentionsAnnotation.class)) {
+                if(rm.getType().equals("_NR")) continue;
+                Map<String, Double> scores = new HashMap<>();
+                for(String label : rm.getTypeProbabilities().keySet())
+                    scores.put(label, rm.getTypeProbabilities().getCount(label));
+                Constituent c1 = createConstituentGivenMention(rm.getEntityMentionArgs().get(0), ta);
+                Constituent c2 = createConstituentGivenMention(rm.getEntityMentionArgs().get(1), ta);
+                Relation r = new Relation(scores, c1, c2);
+                vu.addRelation(r);
+                if(!vu.containsConstituent(c1)) vu.addConstituent(c1);
+                if(!vu.containsConstituent(c2)) vu.addConstituent(c2);
+            }
+        }
+
+        for (CoreMap sentence : document.get(CoreAnnotations.SentencesAnnotation.class)) {
+            for (EntityMention rm : sentence.get(MachineReadingAnnotations.EntityMentionsAnnotation.class)) {
+                Constituent c = createConstituentGivenMention(rm, ta);
+                if(!vu.containsConstituent(c)) vu.addConstituent(c);
+            }
+        }
+
+        ta.addView(viewName, vu);
+    }
+
+    private static Constituent createConstituentGivenMention(EntityMention rm, TextAnnotation ta) {
+        List<CoreLabel> tokens = rm.getSentence().get(CoreAnnotations.TokensAnnotation.class);
+        int extentCharStart = tokens.get(rm.getExtentTokenStart()).beginPosition();
+        int extentCharEnd = tokens.get(rm.getExtentTokenEnd()).endPosition();
+        int headCharStart = tokens.get(rm.getHeadTokenStart()).beginPosition();
+        int headCharEnd = tokens.get(rm.getHeadTokenEnd()).endPosition();
+
+        List<Constituent> extentCons = ta.getView(ViewNames.TOKENS).getConstituentsOverlappingCharSpan(extentCharStart, extentCharEnd);
```

Comment:
it looks like this is doing some heuristic token boundary fixing.  Please add a brief javadoc that gives some idea of behavior.